index.html

```diff
@@ -9064,8 +9050,454 @@
+										<div class="enhancedTextColorPicker" id="sys-text-colorselector">text🖌️</div>
+										<div class="enhancedTextColorPicker instruct-markdown-user" id="sys-speech-colorselector">"speech"🖌️</div>
+										<div class="enhancedTextColorPicker instruct-markdown-user" id="sys-action-colorselector">*action*🖌️</div>
+									</div>
+									<div class="ui-settings-inline instruct-markdown-user">
+										<div style="margin-right:11px; text-align: center;">Code blocks: </div>
+										<div class="enhancedTextColorPicker" id="code-block-background-colorselector">background🖌️</div>
+										<div class="enhancedTextColorPicker" id="code-block-foreground-colorselector">foreground🖌️</div>
+									</div>
+								</div>
+							</div>
+						</div>
+						<div class="popupfooter" style="margin:-10px; width:110%; margin-top: 200px;">
+							<button type="button" class="btn btn-primary" id="btn_settingsaccept" onclick="hideAestheticUISettingsMenu()">OK</button>
+							<button type="button" class="btn btn-primary" id="btn_settingsclose" onclick="hideAestheticUISettingsMenu()">Cancel</button>
+						</div>
+					</div>
+				</div>
+				<div style="background-color: black; flex: 1; padding: 10px;  padding-bottom: 0px; display: flex; flex-direction: column; max-width: 1000px;">
+					<p>Style Preview</p>
+					<div id="aesthetic_text_preview" style="flex: 0; width: 100%; background-color: black; margin: 2px; text-align: left; word-wrap: break-word;"></div>
+				</div>
+				<input type="color" id="colorPicker" style="display: none;">
+				<input type="color" id="colorPicker_background" style="display: none;">
+				<input type="file" id="portraitFileInput" style="display:none" accept="image/*">
+			</div>
+		</div>
+	</div>
 </body>
+
+<style>
+	.ui-settings-inline { font-size: 12px; display:flex; flex-direction: row; }
+	.instruct-settings-input { margin: 0px 2px; font-size:10px; }
+	.instruct-settings-input input { width:40px; height:20px; }
+	#code-block-background-colorselector, #code-block-foreground-colorselector { text-align: center; margin: 0px 5px; }
+	#you-text-colorselector, #you-speech-colorselector, #you-action-colorselector, #AI-text-colorselector, #AI-speech-colorselector, #AI-action-colorselector, #sys-text-colorselector, #sys-speech-colorselector, #sys-action-colorselector { text-align: center; margin: 0px 5px; }
+	#you-bubble-colorselector, #AI-bubble-colorselector, #sys-bubble-colorselector, #you-portrait, #AI-portrait { text-align: center; margin: 0px 10px; border-radius: 1rem; padding: 1px 6px; }
+</style>
+
+<script>	
+const aestheticTextStyleTypes = ['text', 'speech', 'action'];	 // One style per speech type. Could add more later I guess.
+const aestheticTextStyleRoles = ['uniform', 'you', 'AI', 'sys']; // Uniform for when you want all roles use the same styles.
+	
+class AestheticTextStyle {
+    constructor({color = 'rgba(255,255,255, 1)', bold = false, italic = false, opacity = 1} = {}) {
```

Comment:
you may need to polyfill or edit this as this kind of argument initializer only exists on recent browsers. A plain old constructor should be fine - for reference I am testing backwards compatibility via Firefox 45 (earliest which Lite has confirmed support).